crypto/params.c

```diff
@@ -1506,14 +1506,25 @@ static int get_string_ptr_internal(const OSSL_PARAM *p, const void **val,
 
 int OSSL_PARAM_get_utf8_string_ptr(const OSSL_PARAM *p, const char **val)
 {
-    return OSSL_PARAM_get_utf8_ptr(p, val)
-        || get_string_ptr_internal(p, (const void **)val, NULL,
-                                   OSSL_PARAM_UTF8_STRING);
+    int rv;
+
+    ERR_set_mark();
```

Comment:
This is defeating the type checking, I don't think it's a good idea.  We separated these types for a reason.